copycat-file/src/main/java/org/apache/kafka/copycat/file/FileStreamSourceConnector.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.copycat.file;
+
+import org.apache.kafka.copycat.connector.Task;
+import org.apache.kafka.copycat.errors.CopycatException;
+import org.apache.kafka.copycat.source.SourceConnector;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Very simple connector that works with the console. This connector supports both source and
+ * sink modes via its 'mode' setting.
+ */
+public class FileStreamSourceConnector extends SourceConnector {
+    public static final String TOPIC_CONFIG = "topic";
+    public static final String FILE_CONFIG = "file";
+
+    private String filename;
+    private String topic;
+
+    @Override
+    public void start(Properties props) throws CopycatException {
+        filename = props.getProperty(FILE_CONFIG);
+        topic = props.getProperty(TOPIC_CONFIG);
+        if (topic == null || topic.isEmpty())
+            throw new CopycatException("ConsoleConnector configuration must include 'topic' setting");
+        if (topic.contains(","))
+            throw new CopycatException("ConsoleConnector should only have a single topic when used as a source.");
+    }
+
+    @Override
+    public Class<? extends Task> getTaskClass() {
+        return FileStreamSourceTask.class;
+    }
+
+    @Override
+    public List<Properties> getTaskConfigs(int maxTasks) {
```

Comment:
I don't think we can avoid letting the user config this -- they need to be able to specify the maximum level of parallelism they want since each task gets its own thread. But `maxTasks` is meant to just be a suggestion, connectors can always ignore it if they simply cannot generate more tasks.

We could potentially require connectors to specify how many tasks they can generate, but I think this is only useful for a certain class of connectors that simply do not support parallelism at all (i.e. 1 partition in the source data).